base/abstractarray.jl

```diff
@@ -1604,7 +1602,7 @@ function __cat(A, shape::NTuple{M,Int}, catdims, X...) where M
     for x in X
         for i = 1:N
             if concat[i]
-                inds[i] = offsets[i] .+ cat_indices(x, i)
+                inds[i] = offsets[i] .+ parent(cat_indices(x, i))
```

Comment:
What's this about?

Reply:
This strips the offsets and uses linear indexing when computing indices. `parent(r::IdOffsetRange) = r.parent`.

This could be fragile and a little bit tricky, though. Phehaps there's a better way to make things work.

Reply:
I would generally try to stay away from assumptions about how `x` might be structured if at all possible...